osu.Game/Rulesets/UI/Scrolling/ScrollingHitObjectContainer.cs

```diff
@@ -19,6 +19,11 @@ public class ScrollingHitObjectContainer : HitObjectContainer
         private readonly IBindable<double> timeRange = new BindableDouble();
         private readonly IBindable<ScrollingDirection> direction = new Bindable<ScrollingDirection>();
 
+        /// <summary>
+        /// 0 for horizontal scroll, 1 for vertical scroll.
+        /// </summary>
+        private int scrollingAxis => direction.Value == ScrollingDirection.Left || direction.Value == ScrollingDirection.Right ? 0 : 1;
```

Comment:
Maybe better to use `Axes.X`/`Axes.Y` instead? I'm not much of a fan of int magic values.

Reply:
Oh I guess it's done this way for `scrollLength`...

Reply:
I think this could be less magicky if this was a method with a signature of `float getDimensionAlong(ScrollingDirection)` or similar. I *think* that should encompass all usages.

Reply:
`Axes` has `Axes.Both` so it is not the perfect fit.
And the reason I'm using `int` is the `Vector2` indexer.
A new `enum Axis { X, Y }` could be added, but perhaps this should be in osu-framework? The two `Axis` and `Axes` are confusing.

Reply:
I'm fine with a local enum, fwiw. Or even bringing back the old `getBreadth()` stuff, because if I'm honest, after reading this code a bit more to debug the failures this 0/1 stuff is a major readability regression.

Reply:
Actually, never mind. On checking `Direction` [already exists / is public API](https://github.com/ppy/osu-framework/blob/264f29abd5731dcc85c22c4b2d133b4efe41f210/osu.Framework/Graphics/Drawable.cs#L2865-L2869) in framework.